src/mixer/state.ts

```diff
@@ -298,7 +298,7 @@ export const load = (
   if (waveform == null) {
     throw new Error();
   }
-  waveform.innerHTML = ""; // clear previous (ghost) wavesurfer
+  audioEngine.destroyPlayer(player); // clear previous (ghost) wavesurfer and it's media elements.
```

Comment:
Would prefer this called something like `destroyPlayerIfExists`. Or maybe even move it into the `createPlayer function`.

Reply:
Ok. Can't put it in createPlayer because the HTML still needs to be goneski before the loading progress bar so it doesn't break styling. But I'll change the name of it :)